### PR TITLE
Fix nav header accessibility issues

### LINKED ISSUE #6410: [Main menu - Site Admin] account menu item role should be button and collapse attribute properly populated
## Area or Component

Main navigation menu with a site admin role.

## Quick Summary

The account button, which upon clicking displays a submenu, has a role of link and no collapsible aria attribute.

### How to Replicate

Check the DOM.

## Accessibility Violation

[The element's role is missing or is not appropriate for the element's function.
](https://docs.deque.com/issue-help/1.0.0/en/aria-role-missing)

[The element has missing or incorrect states or properties that are necessary for screen reader users to interact with or understand the content conveyed by the element.](https://docs.deque.com/issue-help/1.0.0/en/aria-state-property-missing)

## Action Requested

Set the role to button
Add aria-collapse = true/false depending the state of the submenu.

## Screenshots (optional)
Screenshot 2023-08-24 at 10 29 28 AM

### PR DESCRIPTION
**Pull request by emyl3**
# FRONTEND PULL REQUEST

## Related Issue

Resolves #6410 

## Changes Proposed
Screenshot 2023-09-11 at 23 00 32

- Account icon in header
  - set `role` to `button` instead of `link`
  - add [`aria-expanded`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded) and set to `true/false` depending the state of the submenu
  - add [`aria-controls`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-controls)
- Fix other accessibility issues found when adding jest-axe test to `<Header>` component

## Additional Information
- Adding the jest-axe test to the `<Header>` component revealed that the axe dev tool test didn't run the tests on certain parts of the nav header unless the Account icon button was clicked and expanded.
- Some changes in this PR addresses the issues flagged by the jest-axe tests

## Testing

- deployed on dev2

## Screenshots / Demos

**Before**

Screenshot 2023-09-11 at 22 56 28

**After**

Screenshot 2023-09-11 at 22 55 51

⚠️ The line styling in between the "USER NAME" and "USER ROLE" in the Account dropdown has changed slightly since we cannot use `<hr>` element in a `<ul>` element 

Screenshot 2023-09-11 at 22 55 16

### CODE REVIEW

File: frontend/src/app/commonComponents/ChangeUser.tsx [hunk 1]
```diff
@@ -4,12 +4,14 @@ const ChangeUser = () => {
   }
 
   const getLink = (email: string, name: string) => (
-    <a
-      className="ghost-user-link"
-      href={`/#access_token=SR-DEMO-LOGIN%20${email}`}
-    >
-      Login as {name}
-    </a>
+    <li>
```
Comment (emyl3) [inline frontend/src/app/commonComponents/ChangeUser.tsx:7]: ⚠️ caught by axe-jest, cannot have `<li>` element within `<ul>` parent element

File: frontend/src/app/commonComponents/Dropdown.tsx [hunk 2]
```diff
@@ -91,6 +93,7 @@ const Dropdown: React.FC<Props & SelectProps> = ({
             )}
             name={name}
             id={id}
+            aria-label={ariaLabel}
```
Comment (emyl3) [inline frontend/src/app/commonComponents/Dropdown.tsx:96]: ⚠️ caught by axe-jest, need to have some sort of label for `<select>`

File: frontend/src/app/commonComponents/Header.tsx [hunk 3]
```diff
@@ -281,15 +280,23 @@ const Header: React.FC<{}> = () => {
...
           >
             {deviceType === "desktop" ? item.icon : item.mobileDisplayText}
           </LinkWithQuery>
           {item.hasSubmenu &&
           staffDetailsVisible &&
           deviceType === "desktop" ? (
             <div
+              id={`${deviceType}-${item.dataTestId}-submenu`}
               ref={staffDefailsRef}
-              aria-label="Primary navigation"
```
Comment (emyl3) [inline frontend/src/app/commonComponents/Header.tsx:292]: ⚠️ caught by axe-jest, cannot repeat same aria-label

File: frontend/src/app/commonComponents/ChangeUser.tsx (hunk 1 above)
Comment (fzhao99) [reply frontend/src/app/commonComponents/ChangeUser.tsx:7]: just to clarify, did you mean that we can't have a `<a>` element within a `<ul>`? wanted to make sure I understood for future reference!

**Review by fzhao99 — APPROVED** [review]
lgtm!

File: frontend/src/app/commonComponents/ChangeUser.tsx (hunk 1 above)
Comment (emyl3) [reply frontend/src/app/commonComponents/ChangeUser.tsx:7]: ooo sorry I missed this comment. I think there were only three elements that could be direct children of the `<ul>` element... let me try to see if I can get that error again.

Comment (emyl3) [reply frontend/src/app/commonComponents/ChangeUser.tsx:7]: @fzhao99 it was this error! https://dequeuniversity.com/rules/axe/4.7/list?application=AxeChrome

**Pull request merged**